From 814dc5d972fb451413959441618b48cde08cc61e Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:36:25 -0400
Subject: [ethernet] i40e: fix nvm version and remove firmware report

Message-id: <1409658042-22689-25-git-send-email-sassmann@redhat.com>
Patchwork-id: 90217
O-Subject: [RHEL7.1 PATCH 024/281] i40e: fix nvm version and remove firmware report
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jesse Brandeburg <jesse.brandeburg@intel.com>

The driver needs to use the format that the current NVM
uses when printing the version of the NVM.  It should remain
this way from now on forward.

The driver was reporting when firmware was less than
an expected version number, but this is not a requirement
for the product and we print the firmware number at
init and in ethtool -i output.  Just remove the print.

Change-ID: Ide0b856cd454ebf867610ef9a0d639bb358a4a60
Signed-off-by: Jesse Brandeburg <jesse.brandeburg@intel.com>
Signed-off-by: Catherine Sullivan <catherine.sullivan@intel.com>
Tested-by: Kavindya Deegala <kavindya.s.deegala@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit ff80301efad4818938470326b9879bb960f5e66c)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e.h b/drivers/net/ethernet/intel/i40e/i40e.h
index ba77fca..838b69b 100644
--- a/drivers/net/ethernet/intel/i40e/i40e.h
+++ b/drivers/net/ethernet/intel/i40e/i40e.h
@@ -86,12 +86,12 @@
 
 #define I40E_NVM_VERSION_LO_SHIFT  0
 #define I40E_NVM_VERSION_LO_MASK   (0xff << I40E_NVM_VERSION_LO_SHIFT)
-#define I40E_NVM_VERSION_HI_SHIFT  8
-#define I40E_NVM_VERSION_HI_MASK   (0xff << I40E_NVM_VERSION_HI_SHIFT)
+#define I40E_NVM_VERSION_HI_SHIFT  12
+#define I40E_NVM_VERSION_HI_MASK   (0xf << I40E_NVM_VERSION_HI_SHIFT)
 
 /* The values in here are decimal coded as hex as is the case in the NVM map*/
 #define I40E_CURRENT_NVM_VERSION_HI 0x2
-#define I40E_CURRENT_NVM_VERSION_LO 0x30
+#define I40E_CURRENT_NVM_VERSION_LO 0x40
 
 /* magic for getting defines into strings */
 #define STRINGIFY(foo)  #foo
@@ -489,10 +489,10 @@ static inline char *i40e_fw_version_str(struct i40e_hw *hw)
    "f%d.%d a%d.%d n%02x.%02x e%08x",
    hw->aq.fw_maj_ver, hw->aq.fw_min_ver,
    hw->aq.api_maj_ver, hw->aq.api_min_ver,
-   (hw->nvm.version & I40E_NVM_VERSION_HI_MASK)
-      >> I40E_NVM_VERSION_HI_SHIFT,
-   (hw->nvm.version & I40E_NVM_VERSION_LO_MASK)
-      >> I40E_NVM_VERSION_LO_SHIFT,
+   (hw->nvm.version & I40E_NVM_VERSION_HI_MASK) >>
+   I40E_NVM_VERSION_HI_SHIFT,
+   (hw->nvm.version & I40E_NVM_VERSION_LO_MASK) >>
+   I40E_NVM_VERSION_LO_SHIFT,
    hw->nvm.eetrack);
 
  return buf;
diff --git a/drivers/net/ethernet/intel/i40e/i40e_main.c b/drivers/net/ethernet/intel/i40e/i40e_main.c
index 50d0032..0fc6f9d 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_main.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_main.c
@@ -7981,13 +7981,6 @@ static int i40e_probe(struct pci_dev *pdev, const struct pci_device_id *ent)
 
  err = i40e_init_adminq(hw);
  dev_info(&pdev->dev, "%s\n", i40e_fw_version_str(hw));
- if (((hw->nvm.version & I40E_NVM_VERSION_HI_MASK)
-   >> I40E_NVM_VERSION_HI_SHIFT) != I40E_CURRENT_NVM_VERSION_HI) {
-  dev_info(&pdev->dev,
-    "warning: NVM version not supported, supported version: %02x.%02x\n",
-    I40E_CURRENT_NVM_VERSION_HI,
-    I40E_CURRENT_NVM_VERSION_LO);
- }
  if (err) {
   dev_info(&pdev->dev,
     "init_adminq failed: %d expecting API %02x.%02x\n",
-- 
1.7.1